src/app/pages/credentials/certificates-dash/certificate-list/certificate-list.component.ts

```diff
@@ -167,23 +164,42 @@ export class CertificateListComponent implements OnInit {
   }
 
   doDelete(certificate: Certificate): void {
-    this.matDialog.open(ConfirmForceDeleteCertificateComponent, { data: certificate })
-      .afterClosed()
-      .pipe(
-        filter(Boolean),
-        switchMap((data: { force: boolean }) => {
-          return this.dialogService.jobDialog(
-            this.ws.job('certificate.delete', [certificate.id, data.force]),
-            { title: this.translate.instant('Deleting...') },
-          ).afterClosed();
-        }),
-        this.errorHandler.catchError(),
-        untilDestroyed(this),
-      )
-      .subscribe(() => {
-        this.getCertificates();
-        this.certificateDeleted.emit();
-      });
+    this.dialogService.confirm({
+      title: this.translate.instant('Delete Certificate'),
+      message: this.translate.instant('Are you sure you want to delete "{name}"?', { name: certificate.name }),
+      hideCheckbox: true,
+      secondaryCheckbox: true,
+      buttonColor: 'red',
+      secondaryCheckboxText: this.translate.instant('Force'),
+      buttonText: this.translate.instant('Delete'),
+    }).pipe(
+      filter(
+        (confirmation: boolean | { confirmed: boolean; secondaryCheckbox: boolean }) => {
```

Comment:
Here, below and other files you shouldn't have to do this. `confirm` returns an object when used with `secondaryCheckbox: true`